src/ghci/stdout.rs

```diff
@@ -203,20 +128,17 @@ impl GhciStdout {
     }
 
     #[instrument(skip_all, level = "debug")]
-    async fn show_modules(&mut self, sender: oneshot::Sender<ModuleSet>) -> miette::Result<()> {
+    pub async fn show_modules(&mut self) -> miette::Result<ModuleSet> {
```

Comment:
```suggestion
    pub fn show_modules(&mut self) -> miette::Result<ModuleSet> {
```

Can be synchronous.

Reply:
I think this `async` is still necessary.  There's a `.await` in this function